dom.bs

```diff
@@ -1419,6 +1419,142 @@ that gave folks all the wrong ideas. <a>Events</a> do not represent or cause act
 can only be used to influence an ongoing one.
 
 
+<h2 id=aborting-ongoing-activities>Aborting Ongoing Activities</h3>
+
+Though {{Promise}} objects don't have any built-in aborting mechanism, many APIs using these
+concepts require abort semantics. {{AbortController}} is meant to support these requirements by
+providing an {{AbortController/abort()}} method that toggles the state of a corresponding
+{{AbortSignal}} object. The API which wishes to support aborting can accept such a {{AbortSignal}},
+and use its state to determine how (not) to proceed.
+
+APIs that rely upon {{AbortController}} are encouraged to respond to {{AbortController/abort()}} by
+rejecting any unsettled {{Promise}} with a new {{DOMException}} with [=error name=]
+"{{AbortError}}".
+
+<div class=example id=aborting-ongoing-activities-example>
+ For example, a hypothetical <code>doAmazingness({ ... })</code> method could accept a
+ {{AbortSignal}} object in order to support aborting as follows:
+
+ <pre class=lang-javascript>
+  const controller = new AbortController();
+  const signal = controller.signal;
+
+  startSpinner();
+
+  doAmazingness({ ..., signal })
+    .then(result => ...)
+    .catch(err => {
+      if (err.name == 'AbortError') return;
+      showUserErrorMessage();
+    })
+    .then(() => stopSpinner());
+
+  ...
+
+  controller.abort();
+ </pre>
+
+ <code>doAmazingness</code> could be implemented as follows:
+
+ <pre class=lang-javascript>
+  function doAmazingness({signal}) {
+    return new Promise((resolve, reject) => {
+      // Begin doing amazingness, and call resolve(result) when done.
+      // But also, watch for signals:
+      signal.addEventListener('abort', () => {
+        // Stop doing amazingness, and:
+        reject(new DOMException('Aborted', 'AbortError'));
+      });
+    });
+  }
+ </pre>
+
+ APIs that require more granular control could extend both {{AbortController}} and {{AbortSignal}}
+ according to their needs.
+</div>
+
+<h3 id=interface-abortcontroller>Interface {{AbortController}}</h3>
+
+<pre class="idl">
+[Constructor(), Exposed=(Window,Worker)]
+interface AbortController {
+  [SameObject] readonly attribute AbortSignal signal;
+
+  void abort();
```

Comment:
I agree with your choice to return `void` here. In particular, I expect that returning ["a Promise that is settled when cancellation has completed"](https://github.com/rbuckton/prex/blob/master/docs/cancellation.md#sourcecancel) will encourage developers to miss the fact that "an API listening to a signal may choose to ignore it" and so write bugs.